Automattic/wp-calypso
# Remove reader/comments component and switch my-sites/posts over to the newer block

The `reader/comments` component has been superceded by the new comments block, `blocks/comments`.

The Reader itself is already using the new block, but My Sites > Posts still uses the older version.

This PR removes the old component and switches My Sites > Posts to use the new block.

<img width="763" alt="screen shot 2017-01-03 at 16 09 41" src="https://cloud.githubusercontent.com/assets/17325/21600366/28db5be4-d1cf-11e6-9f91-97112bdb70bd.png">

### To test

Head to http://calypso.localhost:3000/posts/my and ensure that comment threads appear correctly.

### Todo

- [x] switch `my-sites/posts/post.jsx` over to use new comments block
- [x] tidy up `!important` styles in `blocks/comments/style.scss`
- [x] remove `reader/comments` component completely
- [x] turn off "x comments" header for My Sites
- [x] tweak date spacing

Fixes https://github.com/Automattic/wp-calypso/issues/9705.
